jerry-debugger/src/lib/__tests__/cdt-proxy-http.test.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright JS Foundation and other contributors, http://js.foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { onHttpRequest } from '../cdt-proxy-http';
+
+describe('onHttpRequest', () => {
+  // common setup
+  const proxy = {
+    host: '127.0.0.1',
+    port: 9229,
+    jsfile: 'foo/bar/baz.js',
+  };
+  const request = {
+    method: 'GET',
+    url: '',
+  };
+  const response = {
+    statusCode: 200,
+    setHeader: jest.fn(),
+    end: jest.fn(),
+    write: jest.fn(),
+  };
+
+  // clean up before each test
+  beforeEach(() => {
+    request.method = 'GET';
+    response.statusCode = 200;
+  });
+
+  it('responds to POST with 405', () => {
+    request.method = 'POST';
+    onHttpRequest.call(proxy, request, response);
+    expect(response.statusCode).toEqual(405);
+    expect(response.end).toBeCalled();
+  });
+
+  it('responds to unexpected path with 404', () => {
+    request.url = '/foo';
+    onHttpRequest.call(proxy, request, response);
+    expect(response.statusCode).toEqual(404);
+    expect(response.end).toBeCalled();
+  });
+
+  it('responds to version query with JSON', () => {
+    request.url = '/json/version';
+    onHttpRequest.call(proxy, request, response);
+    expect(response.statusCode).toEqual(200);
+    expect(response.end).toBeCalled();
+    expect(response.write).toHaveBeenCalled();
+
+    const obj = JSON.parse(response.write.mock.calls[0][0]);
+    expect(obj['Browser']).toBeDefined();
+    expect(obj['Protocol-Version']).toBeDefined();
+  });
+
+  let saveJSON = '';
```

Comment:
Tests should not share state. The order in which the tests are run is up to the test runner. Also, in a future version, it could be that tests run in true isolation.